modules/andes-core/broker/src/main/java/org/wso2/andes/amqp/AMQPUtils.java

```diff
@@ -387,18 +389,30 @@ public static InboundExchangeEvent createAndesExchange(Exchange exchange) {
      * @param routingKey routing key
      * @return InboundBindingEvent binding event that wrap AndesBinding
      */
-    public static InboundBindingEvent createAndesBinding(Exchange exchange, AMQQueue queue, AMQShortString routingKey) {
+    public static InboundBindingEvent createAndesBinding(Exchange exchange, AMQQueue queue, AMQShortString routingKey)
+            throws AMQInternalException {
         /**
          * we are checking the type of exchange to avoid the confusion
          * <<default>> exchange is actually a direct exchange. No need to keep two bindings
          */
         //TODO: extend to other types of exchanges
         String exchangeName = "";
+        log.info("Create andes binding for AMQ Queue : "+ queue+ ", Routing Key is "+routingKey);
+
         if (exchange.getType().equals(DirectExchange.TYPE)) {
             exchangeName = DirectExchange.TYPE.getDefaultExchangeName().toString();
         } else if (exchange.getType().equals(TopicExchange.TYPE)) {
             exchangeName = TopicExchange.TYPE.getDefaultExchangeName().toString();
         }
+
+        //Validating topic names by validating the routing key
+        if(!Pattern.matches("^(((\\*){1}|\\w+)+((\\.\\w+)*(\\.\\*)*)*(\\.\\#)?)|((\\#){1})$" , routingKey)){
```

Comment:
Since this pattern is used multiple times, compiling it once and reusing it will be more efficient than invoking matches each time. 

E.g.

    Pattern p = Pattern.compile("a*b");
    Matcher m = p.matcher("aaaaab");
    boolean b = m.matches()